internal/corazawaf/rulegroup.go

```diff
@@ -154,3 +159,15 @@ func NewRuleGroup() RuleGroup {
 		rules: []*Rule{},
 	}
 }
+
+type transformationKey struct {
```

Comment:
@M4tteoP The main difference with your initial approach is using a struct instead of string for the key, which I failed to communicate well offline :) There is significant cost including allocation to concatenate strings, while filling a struct is much faster and all on the stack so no allocation

Reply:
Rag, Many thanks taking over it while explaining how you did it! Very appreciated.